app/modules/assets/models.py

```diff
@@ -16,6 +18,74 @@
 log = logging.getLogger(__name__)  # pylint: disable=invalid-name
 
 
+class AssetSync(ACMSyncMixin):
+    """
+    Class to build up the required data to create an asset,
+    relies upon the ACMSyncMixin calling acm_sync_complete to do so
+    """
+
+    # fmt: off
+    # Name of the module, used for knowing what to sync i.e asset.list, asset.data
+    ACM_NAME = 'assets'
+
+    ACM_LOG_ATTRIBUTES = [
+        'name',
+    ]
+
+    ACM_ATTRIBUTE_MAPPING = {
+        # TODO no idea at the moment as have not yet found the correct
+        # incantation to see this kind of information from WBIA
+        # Ignored
+        #'id'                    : None,
+        #'created'               : None,
+        #'modified'              : None,
+
+        # # Attributes
+        #'name'                  : 'title',
+        #'url'                   : 'website',
+        #'version'               : 'version',
+
+        # # Functions
+        #'annotations'           : '_process_annotations',
+        #'createdDate'           : '_process_created_date',
+        #'modifiedDate'          : '_process_modified_date',
+    }
+    # fmt: on
+
+    def __init__(self, guid):
+        super().__init__()
+        self.guid = uuid.UUID(guid)
+        self.annotations = []
+
+    @classmethod
+    def ensure_acm_obj(cls, guid):
+        """
+        Doesn't ensure that an object exists, creates a new one to perform the sync
+        """
+        assetSync = AssetSync(guid=guid)
+
+        return assetSync
+
+    def acm_sync_complete(self):
+        # TODO update existing asset or create the new asset from the populated data in AssetSync
```

Comment:
Thoughts on this? 
Asset has loads of columns that are nullable=False so we can't create an empty one for population.
Also where does the Submission data come from? AFAIK ACM knows nothing about the submission and neither does EDM.